src/onchain/mod.rs

```diff
@@ -87,7 +87,7 @@ where
         return Ok(None);
     }
 
-    let lsp_fees = swap.calculate_lsp_fee_for_amount(amount.msats)?;
+    let lsp_fees = swap.retrieve_lsp_fee_for_amount(amount.msats)?;
```

Comment:
Not sure why the lsp_fee is being requested twice. Can this statement be removed?